src/main/java/seedu/duke/Messages.java

```diff
@@ -122,6 +122,18 @@ public class Messages {
             + LINE_BREAK
             + "\nNote: Only unit type labels (Singapore-Based) are accepted by the program";
 
+    public static final String MESSAGE_ADDRESS_FORMAT_UNIT_TYPE_MISMATCH = "OOPS!!! There seem to be a mismatch"
+            + " between address format and unit type provided.\n"
+            + "Certain unit types will require specific address format. Please refer to the description below:\n"
+            + "  1. Unit type with <LP> must not have #[unit level]-[unit number] in address.\n"
+            + "     Format:\n"
+            + "       [BLOCK NUMBER] [STREET NAME], S[POSTAL CODE]\n"
+            + "  2. Unit type without <LP> must have #[unit level]-[unit number] in address.\n"
+            + "     Format:\n"
+            + "       [BLOCK NUMBER] [STREET NAME] #[unit level]-[unit number], S[POSTAL CODE]\n"
+            + "       [BLOCK NUMBER] [STREET NAME] #[unit level]-[unit number] [building name], S[POSTAL CODE]\n"
+            + "Note: HDB Terrace House (special case) is not restricted by any format.";
+
```

Comment:
Good elaboration on the error message.

Reply:
Agree, clear message with good formatting